File: week-1(Pset-1)/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int get_positive_int(string prompt);

int main(void)
{
    int number = get_positive_int("Height: ");

    for (int height = 0; height < number; height++) // height and number of pyramid lines
    {
        for (int spaces = number - height - 1; spaces > 0; spaces--) // print spaces
        {
            printf(" ");
        }
        for (int blocks = 0; blocks <= height; blocks++) // left pyramid
        {
            printf("#");
        }
        printf("  ");
        for (int blocks = 0; blocks <= height; blocks++) // right pyramid
        {
            printf("#");
        }
        printf("\n");
    }
}

int get_positive_int(string prompt) // def func.
{
    int number;
    do
    {
        number = get_int("%s", prompt);
    }
    while (number < 1 || number > 8); // range between 1 to 8
    return number;
}

File: week-5(Pset-5)/sentimental-mario-more/mario.py
def get_positive_int(prompt):
    while True:
        try:
            # get the user imput and convert to int
            number = int(input(prompt))
            # Check if the number is between 1 and 8 (inclusive)
            if 1 <= number <= 8:
                return number
            else:
                print("Invalid input. Please enter a number between 1 and 8.")
        except ValueError:
            print("Invalid input. Please enter a valid Integer.")


def main():
    # Get the height of the pyramid from the user
    height = get_positive_int("Height: ")

    for i in range(height):

        for j in range(height - i - 1):
            print(" ", end="")

        for j in range(i + 1):
            print("#", end="")

        # print gap between the two pyramids
        print("  ", end="")

        for j in range(i + 1):
            print("#", end="")

        print()


# execute the main func if this script is run directly
if __name__ == "__main__":
    main()
